src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlSettingsAction.java

```diff
@@ -0,0 +1,122 @@
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.rest.RestStatus.INTERNAL_SERVER_ERROR;
+
+/**
+ * Interface to manage opendistro.sql.* cluster settings
+ * All non-sql settings are ignored.
+ * Any non-transient and non-persistent settings are ignored.
+ */
+public class RestSqlSettingsAction extends BaseRestHandler {
+    private static final Logger LOG = LogManager.getLogger(RestSqlSettingsAction.class);
+
+    private static final String PERSISTENT = "persistent";
+    private static final String TRANSIENT = "transient";
+    private static final String SQL_SETTINGS_PREFIX = "opendistro.sql.";
+
+    /**
+     * API endpoint path
+     */
+    public static final String SETTINGS_API_ENDPOINT = "/_opendistro/_sql/settings";
+
+    public RestSqlSettingsAction(Settings settings, RestController restController) {
+        super();
+        restController.registerHandler(RestRequest.Method.POST, SETTINGS_API_ENDPOINT, this);
+    }
+
+    @Override
+    public String getName() {
+        return "sql_settings_action";
+    }
+
+    /**
+     * @see org.elasticsearch.rest.action.admin.cluster.RestClusterUpdateSettingsAction
+     */
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        LogUtils.addRequestId();
+        final ClusterUpdateSettingsRequest clusterUpdateSettingsRequest = Requests.clusterUpdateSettingsRequest();
+        clusterUpdateSettingsRequest.timeout(request.paramAsTime("timeout", clusterUpdateSettingsRequest.timeout()));
+        clusterUpdateSettingsRequest.masterNodeTimeout(
+                request.paramAsTime("master_timeout", clusterUpdateSettingsRequest.masterNodeTimeout()));
+        Map<String, Object> source;
+        try (XContentParser parser = request.contentParser()) {
+            source = parser.map();
+        }
+
+        try {
+            if (source.containsKey(TRANSIENT)) {
+                clusterUpdateSettingsRequest.transientSettings((Map) source.get(TRANSIENT));
+            }
+            if (source.containsKey(PERSISTENT)) {
+                clusterUpdateSettingsRequest.persistentSettings((Map) source.get(PERSISTENT));
+            }
+
+            // filter out all non-sql settings
+            clusterUpdateSettingsRequest.transientSettings(
+                    filterSettings(clusterUpdateSettingsRequest.transientSettings())
+            );
+            clusterUpdateSettingsRequest.persistentSettings(
+                    filterSettings(clusterUpdateSettingsRequest.persistentSettings())
+            );
+
+            return channel -> client.admin().cluster().updateSettings(
```

Comment:
Does this block operation on transport thread?

Reply:
Let me confirm.

Reply:
Verified that this is not blocking operation on Transport thread, in two ways:

1. Our integration test cluster already has assertions enabled (as confirmed with @dai-chen ) and all settings related integration tests are passing.

2. Manually tested by adding `-ea` flag in `config/jvm.options`, and changed settings via Kibana Devtools console.